Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -55,24 +56,30 @@
 namespace GCAdapter
 {
 #if GCADAPTER_USE_LIBUSB_IMPLEMENTATION
+
+constexpr unsigned int USB_TIMEOUT_MS = 16;
+
 static bool CheckDeviceAccess(libusb_device* device);
 static void AddGCAdapter(libusb_device* device);
 static void ResetRumbleLockNeeded();
 #endif
+
 static void Reset();
 static void Setup();
-static void Read();
-static void Write();
+static void ProcessInputPayload(const u8* data, std::size_t size);
+static void ReadThreadFunc();
+static void WriteThreadFunc();
 
 #if GCADAPTER_USE_LIBUSB_IMPLEMENTATION
-enum
+enum class AdapterStatus
 {
-  NO_ADAPTER_DETECTED = 0,
-  ADAPTER_DETECTED = 1,
+  NotDetected,
+  Detected,
+  Error,
 };
 
-// Current adapter status: detected/not detected/in error (libusb error codes are negative)
-static std::atomic<int> s_status = NO_ADAPTER_DETECTED;
+static std::atomic<AdapterStatus> s_status = AdapterStatus::NotDetected;
+static std::atomic<libusb_error> s_adapter_error;
```

Comment:
IMO this should be initialized to `LIBUSB_SUCCESS` (and the same should happen on line 429 of `Init()`).

Reply:
I can do that.